src/main/scala/diplomaticobjectmodel/logicaltree/RocketLogicalTreeNode.scala

```diff
@@ -14,20 +14,20 @@ import freechips.rocketchip.tile.{RocketTileParams, TileParams, XLen}
  * The data memory subsystem is assumed to be a DTIM if and only if deviceOpt is
  * a Some(SimpleDevice), as a DCache would not create a Device.
  */
-class DCacheLogicalTreeNode(deviceOpt: Option[SimpleDevice], params: DCacheParams) extends LogicalTreeNode(() => deviceOpt) {
+class DCacheLogicalTreeNode(memories: () => Seq[OMSRAM], deviceOpt: Option[SimpleDevice], params: DCacheParams) extends LogicalTreeNode(() => deviceOpt) {
```

Comment:
For consistency, can you place this new `memories` parameter at the end of the parameters list? This would not only make it match the order of the `OMCaches.dcache()` function parameters, but it would also make it consistent with every other change you made to function signatures here.